docs/concepts/jarsigning.md

```diff
@@ -0,0 +1,73 @@
+Signing a JAR
+=============
+
+Java allows developers to sign their jar files with a signature. However, these signatures are not designed 
+to be used as security and shouldn't be used this way. Signatures are used as sanity checks, so that developers
+are able to check if they are running their own un-edited code.
+
+!!! note
+
+	Once again keep in mind that this system is not a security measure. With enough malicious intend it can be circumvented.
+	
+Creating a keystore
+-------------------
+A **keystore** is a private database file which holds the required information to successfully sign the jar.
+To sign a jar, you need both a public and a private key. The public key will be later used to check if the
+jar is correctly signed.
+
+To generate a key, execute the following command and follow the instructions given by the keytool.
+The key should be **SHA-1 encoded**.
+```shell
+keytool -genkey -alias signFiles -keystore examplestore.jks
+```
+The `keytool` is part of the Java Development Kit and can be found in the underlying `bin` directory.
+The `alias signFiles` indicates that the alias should be used in future to refer to the keystore entry and
+`-keystore examplestore.jks` means that the keystore will be saved to the file `examplestore.jks`.
+
+### Get the public key
+To gather the public key required by Forge, execute the following command:
+```shell
+keytool -list -alias signFiles -keystore examplestore.jks
+```
+Now copy the public key and remove all colons and change all uppercase letters to lowercase to ensure
+that FML can work with the key.
+
+### Checking at runtime
+To allow FML to compare the keys, add the public key to the `certificateFingerprint` argument in the `@Mod` annotation.
+
+When FML detects that the keys don't match it will fire the mod-lifecycle event `FMLFingerprintViolationEvent` How to 
+handle this key mismatch is up to the developer.
+
+- `event.isDirectory()` - Returns true when the mod runs in development environment.
+- `event.getSource()` - Returns the file with the key mismatch.
+- `event.getExpectedFingerprint()` - Returns the public key.
+- `event.getFingerprints()` - Returns all public keys found.
+
+!!! note
+
+	The event runs before any of the Mod's code is executed.
```

Comment:
This doesn't really make much sense. It is also probably incorrect for static initializers and that really early FML phase event.